client/my-sites/plans-features-main/hooks/data-store/use-pricing-meta-for-grid-plans.ts

```diff
@@ -91,15 +102,31 @@ const usePricingMetaForGridPlans: UsePricingMetaForGridPlans = ( {
 
 				// raw prices for current site's plan
 				if ( selectedSiteId && currentSitePlanSlug === planSlug ) {
-					const monthlyPrice = getSitePlanRawPrice( state, selectedSiteId, planSlug, {
+					let monthlyPrice = getSitePlanRawPrice( state, selectedSiteId, planSlug, {
 						returnMonthly: true,
 						returnSmallestUnit: true,
 					} );
-					const yearlyPrice = getSitePlanRawPrice( state, selectedSiteId, planSlug, {
+					let yearlyPrice = getSitePlanRawPrice( state, selectedSiteId, planSlug, {
 						returnMonthly: false,
 						returnSmallestUnit: true,
 					} );
 
+					/**
+					 * Ensure the spotlight plan shows the price with which the plans was purchased.
+					 */
+					if ( purchasedPlan ) {
+						const isMonthly = purchasedPlan.billPeriodDays === PLAN_MONTHLY_PERIOD;
+						const isYearly = purchasedPlan.billPeriodDays === PLAN_ANNUAL_PERIOD;
+
+						if ( isMonthly && monthlyPrice !== purchasedPlan.priceInteger ) {
+							monthlyPrice = purchasedPlan.priceInteger;
+							yearlyPrice = Math.floor( purchasedPlan.priceInteger * 12 );
+						} else if ( isYearly && yearlyPrice !== purchasedPlan.priceInteger ) {
+							monthlyPrice = Math.floor( purchasedPlan.priceInteger / 12 );
```

Comment:
Thanks for the updates. It looks like it's in the right place now :-)

Curious if we should be using the same or similar logic to `calculateMonthlyPriceForPlan`/`calculateMonthlyPrice` for monthly/yearly conversions (parse float, fixed-point conversion, etc)

Reply:
Good idea. I added that as well, now.